dbms/src/Core/Field.h

```diff
@@ -122,7 +122,7 @@ class DecimalField
         }
         if (cnt == 0)
             cnt = 1;
-        return cnt;
+        return std::max(cnt, scale);
```

Comment:
why?

Reply:
For Decimal(5,2) type, and the DecimalField is 0, then the prec is 1, which is less than the scale.